extensions/amp-graphiq/0.1/amp-graphiq.js

```diff
@@ -0,0 +1,165 @@
+ *   width="320"
+ *   height="392"
+ *   data-href="https://www.graphiq.com/vlp/ekkg7oSEwYZ">
+ * </amp-graphiq>
+ * </code>
+ */
+
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {setStyles} from '../../../src/style';
+import {removeElement} from '../../../src/dom';
+import {user} from '../../../src/log';
+import {tryParseJson} from '../../../src/json';
+import {isObject} from '../../../src/types';
+import {listen} from '../../../src/event-helper';
+
+class AmpGraphiq extends AMP.BaseElement {
+
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /** @private {?Element} */
+    this.iframe_ = null;
+
+    /** @private {!string} */
+    this.widgetId_ = '';
+
+    /** @private {!string} */
+    this.href_ = '';
+
+    /** @private {?Function} */
+    this.unlistenMessage_ = null;
+  }
+ /**
+  * @param {boolean=} opt_onLayout
+  * @override
+  */
+  preconnectCallback(opt_onLayout) {
+    this.preconnect.url('https://w.graphiq.com', opt_onLayout);
+  }
+
+  /** @override */
+  renderOutsideViewport() {
+    return false;
+  }
+
+  /** @override */
+  buildCallback() {
+    this.widgetId_ = user().assert(
+        this.element.getAttribute('data-widget-id'),
+        'The data-widget-id attribute is required for <amp-graphiq> %s',
+        this.element);
+    this.href_ = this.element.getAttribute('data-href') ||
+        'https://www.graphiq.com/vlp/' + encodeURIComponent(this.widgetId_);
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  layoutCallback() {
+    const iframe = this.element.ownerDocument.createElement('iframe');
+    this.iframe_ = iframe;
+    const initialHeight = this.element.getAttribute('height') || '';
+    const initialWidth = this.element.getAttribute('width') || '';
+
+    this.unlistenMessage_ = listen(
+      this.win,
+      'message',
+      this.handleGraphiqMessages_.bind(this)
+    );
+
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('allowtransparency', 'true');
+    //Add title to the iframe for better accessibility.
+    iframe.setAttribute('title', 'Graphiq: ' +
+        this.element.getAttribute('alt') || '');
+    iframe.setAttribute('height', initialHeight);
+    iframe.setAttribute('width', initialWidth);
+    iframe.src = 'https://w.graphiq.com/w/' + encodeURIComponent(this.widgetId_) +
+        '?data-width=' + encodeURIComponent(initialWidth) +
+        '&data-height=' + encodeURIComponent(initialHeight) +
+        '&data-href=' + encodeURIComponent(this.href_);
+
+    this.applyFillContent(iframe);
+    this.element.appendChild(iframe);
+    setStyles(iframe, {
+      'opacity': 0,
+    });
+    return this.loadPromise(iframe).then(() => {
+      this.getVsync().mutate(() => {
+        setStyles(iframe, {
+          'opacity': 1,
+        });
+      });
+    });
+  }
+
+  /** @private */
+  handleGraphiqMessages_(event) {
```

Comment:
Interesting - my (flawed) understanding was that `resize` was beaconing out a different value than `embed-size`.  You're right that there'd be no reason to bring `resize` back if the two will always be the same.  

But is it possible to have Graphiq beacon the actual content height rather than the configured height, either through `resize` or some other new message?  In that case we'd be able to display a full widget without scrollbars, which would be awesome.